### PR TITLE
Encourage true/false for module documentation

### PR DESCRIPTION
**Pull request by kristianheljas**
##### SUMMARY

Make ansible-lint happy and conform to the community decision.

CC @samccann 

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by samccann — APPROVED** [review]

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 1]
```diff
@@ -163,7 +163,7 @@ All fields in the ``DOCUMENTATION`` block are lower-case. All fields are require
     * Ensure that the default value in the docs matches the default value in the code.
     * The default field must not be listed as part of the description, unless it requires additional information or conditions.
     * If the option is a boolean value, you can use any of the boolean values recognized by Ansible:
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst]: I don't think this colon makes any sense.
```suggestion
    * If the option is a boolean value, you can use any of the boolean values recognized by Ansible
```

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 2]
```diff
@@ -163,7 +163,7 @@ All fields in the ``DOCUMENTATION`` block are lower-case. All fields are require
...
-      (such as true/false or yes/no).  Choose the one that reads better in the context of the option.
+      (such as true/false or yes/no).  Document booleans as true/false for consistency and compatibility with ansible-lint.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst]:
```suggestion
      (such as ``true``/``false`` or ``yes``/``no``).  Document booleans as ``true``/``false`` for consistency and compatibility with ansible-lint.
```

**Review by felixfontein — COMMENTED** [review]
Some general improvements for that bullet point, that aren't specific to this change. (The change itself looks good!)

**Review by felixfontein — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @kristianheljas for the fix and @felixfontein for the review!

**Pull request merged**